chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -104,6 +112,74 @@ pub fn validate_prepared_transactions(
     )
 }
 
+struct StateWitnessBlockRange {
+    /// Blocks from the last new chunk (exclusive) to the parent block
+    /// (inclusive).
+    blocks_after_last_chunk: Vec<Block>,
+    /// Blocks from the last last new chunk (exclusive) to the last new chunk
+    /// (inclusive).
+    blocks_after_last_last_chunk: Vec<Block>,
+    last_chunk_shard_id: ShardId,
+    last_chunk_shard_index: ShardIndex,
+}
+
+fn get_state_witness_block_range(
+    state_witness: &ChunkStateWitness,
+    store: &ChainStore,
+    epoch_manager: &dyn EpochManagerAdapter,
```

Comment:
min nit: It's nice to keep the arguments from most general to most specific. In this case I would make it 
store,
epoch manager,
state witness,

It's nice to have a convention and this one is as good as any I guess.